scheduler/pkg/store/experiment/db.go

```diff
@@ -73,12 +75,10 @@ func (edb *ExperimentDBManager) restore(startExperimentCb func(*Experiment) erro
 					return err
 				}
 				experiment := CreateExperimentFromRequest(&snapshot)
-				if err != nil {
-					return err
-				}
 				err = startExperimentCb(experiment)
 				if err != nil {
-					return err
+					// skip restoring the experiment if the callback returns an error
```

Comment:
May I suggest we slightly alter the comment to avoid confusion: the code doesn't skip anything, it simply swallows the error and logs a warning. If we would have bubbled the error up, that would end up stopping the scheduler with a Fatal error in main().

I was thinking of something like: "If the callback fails, do not bubble the error up but simply log it as a warning. The experiment restore is skipped instead of the scheduler failing due to the returned error."